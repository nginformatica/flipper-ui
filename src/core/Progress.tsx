import {
    CircularProgress as MuiCircularProgress,
    LinearProgress as MuiLinearProgress
} from '@material-ui/core'
import React, { FC } from 'react'
import { DefaultProps } from './types'

interface ILinear {
    color?: 'primary' | 'secondary'
    valueBuffer?: number
    variant?: 'determinate' | 'indeterminate' | 'buffer' | 'query'
    classes?: {
        root?: string
        colorPrimary?: string
        barColorPrimary?: string
        barColorSecondary?: string
    }
}

interface ICircular {
    size?: string | number
    color?: 'primary' | 'secondary' | 'inherit'
    variant?: 'determinate' | 'indeterminate' | 'static'
}

interface ProgressProps extends DefaultProps {
    value?: number
    linear?: boolean
}

const Progress: FC<ProgressProps & ICircular & ILinear> = ({
    linear,
    style = {},
    margin,
    padding,
    ...otherProps
}) =>
    linear ? (
        <MuiLinearProgress
            style={{ margin, padding, ...style }}
            {...otherProps}
        />
    ) : (
        <MuiCircularProgress
            style={{ margin, padding, ...style }}
            {...otherProps}
        />
    )

export default Progress
