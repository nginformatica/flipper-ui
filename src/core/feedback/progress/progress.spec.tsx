import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Progress from '.'

describe('Progress', () => {
    it('should render circular', () => {
        render(<Progress />)

        const progress = screen.getByRole('progressbar')

        expect(progress.classList).toContain('MuiCircularProgress-root')
        expect(progress.firstElementChild?.classList).toContain(
            'MuiCircularProgress-svg'
        )
    })

    it('should render linear', () => {
        render(<Progress linear />)

        const progress = screen.getByRole('progressbar')

        expect(progress.classList).toContain('MuiLinearProgress-root')
        expect(progress.firstElementChild?.classList).toContain(
            'MuiLinearProgress-bar'
        )
    })

    it('should match snapshot', () => {
        const { container: circular } = render(<Progress />)
        const { container: linear } = render(<Progress linear />)

        expect(circular).toMatchSnapshot()
        expect(linear).toMatchSnapshot()
    })
})
