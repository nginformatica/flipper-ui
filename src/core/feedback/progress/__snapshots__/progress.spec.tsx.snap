// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should match snapshot 1`] = `
<div>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;

exports[`Progress should match snapshot 2`] = `
<div>
  <span
    class="MuiLinearProgress-root makeStyles-root-17 makeStyles-root-25 MuiLinearProgress-colorPrimary makeStyles-colorPrimary-18 makeStyles-colorPrimary-26 MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary makeStyles-barColorPrimary-19 makeStyles-barColorPrimary-27 MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary makeStyles-barColorPrimary-19 makeStyles-barColorPrimary-27 MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
    />
  </span>
</div>
`;
